lametro/models.py

```diff
@@ -166,6 +166,64 @@ def board_report(self):
 
         return br
 
+    @property
+    def actions_and_agendas(self):
+        '''
+        Return list of dictionaries, each representing an action or an agenda on
+        which the given bill appears, in the format:
+
+        {
+            'date': <datetime>,
+            'description': <str>,
+            'event': <LAMetroEvent>,
+            'organization': <Organization>
+        }
+        '''
+        actions = self.actions.all()
+        events = LAMetroEvent.objects.filter(agenda__related_entities__bill=self)
+
+        data = []
+
+        for action in actions:
+            event = LAMetroEvent.objects.get(\
+                participants__entity_type='organization',\
+                participants__name=action.organization,
+                start_time__date=action.date)
+
+            
+            action_dict = {
+                'date': action.date_dt,
+                'description': action.description,
+                'event': event,
+                'organization': action.organization
+            }
+            
+            data.append(action_dict)
+
+        for event in events:
+            # Use a description of "SCHEDULED"
+            org = event.participants.first()
+            event_dict = {
+                'date': event.start_time.date(),
+                'description': 'SCHEDULED',
+                'event': event,
+                'organization': org
+            }
+
+            data.append(event_dict)
+
+        scheduled_first = []
+        for d in data:
+            if d['description'] == 'SCHEDULED':
+                scheduled_first.append(d)
+        for d in data:
+            if d['description'] != 'SCHEDULED':
+                scheduled_first.append(d)
+
+        sorted_data = sorted(scheduled_first, key=lambda x: x['date'])
```

Comment:
This is clever, but I worry that the first sort won't always persist when we apply the second. What if we used a tuple to sort the data, instead? Here's an example of how tuple sorts work in Python:

```python
>>> sorted([(1, 'A'), (1, 'a'), (0, 'A'), (0, 'AP')])
[(0, 'A'), (0, 'AP'), (1, 'A'), (1, 'a')]
```

You can see that Python sorts are [ascii-betical](https://en.wikipedia.org/w/index.php?title=ASCII&redirect=no#Order), i.e., uppercase letters come before lowercase letters. 

Would something like this do the trick?

```python
lambda x: (x['date'], x['description'].upper() if x['description'] == 'SCHEDULED' else x['description'].lower()
```